stackrox/stackrox
# [ROX-19482] : Add data model changes, conversion and validation support for custom email subject and body

- Added fields to support custom email subject and body in the storage and API data models.
- Added validations for the custom subject and body when creating/updating report configuration.
- Updated conversions between storage and API models to convert custom subject and body.

## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist

If any of these don't apply, please comment below.
This change does not require any special upgrade steps. Change log and documentation will be added in a separate PR.

## Testing Performed

### Here I tell how I validated my change

TODO(replace-me)
Unit tests

Manual tests by making API calls to 
1) create a report configuration with custom subject and body, 
2) getting the report configuration and verifying that the custom subject and body are set.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.
